src/index.ts

```diff
@@ -117,9 +114,9 @@ export class Project {
       keywords: pkg.keywords || [],
     });
     if (files) {
-      this.files = { ...defaultFiles, ...files };
+      this.files = deepmerge({}, { ...defaultFiles, ...files });
     } else {
-      this.files = defaultFiles;
+      this.files = deepmerge({}, defaultFiles);
```

Comment:
An Object.assign or splat here was the only thing that's actually needed to address the test case.

The deep merging is fine but not really related to the bug and not covered by the tests.